mindmeld/auto_annotator.py

```diff
@@ -0,0 +1,709 @@
+        """ Initializes an annotator."""
+        self.app_path = app_path
+        self.config = get_auto_annotator_config(app_path=app_path, config=config)
+        self._resource_loader = (
+            resource_loader or ResourceLoader.create_resource_loader(app_path)
+        )
+        self.annotate_file_entities_map = self._get_file_entities_map(
+            action=AnnotatorAction.ANNOTATE
+        )
+
+    def _get_file_entities_map(self, action: AnnotatorAction):
+        """ Creates a dictionary that maps file paths to entities given
+        regex rules defined in the config.
+
+        Args:
+            action (str): Can be "annotate" or "unannotate". Used as a key
+                to access a list of regex rules in the config dictionary.
+
+        Returns:
+            file_entities_map (dict): A dictionary that maps file paths in an
+                App to a list of entities.
+        """
+        all_file_paths = self._resource_loader.get_all_file_paths()
+        file_entities_map = {path: [] for path in all_file_paths}
+
+        if action == AnnotatorAction.ANNOTATE:
+            rules = self.config[AnnotatorAction.ANNOTATE.value]
+        elif action == AnnotatorAction.UNANNOTATE:
+            rules = self.config[AnnotatorAction.UNANNOTATE.value]
+
+        for rule in rules:
+            pattern = Annotator._get_pattern(rule)
+            compiled_pattern = re.compile(pattern)
+            filtered_paths = self._resource_loader.filter_file_paths(
+                compiled_pattern=compiled_pattern, file_paths=all_file_paths
+            )
+            for path in filtered_paths:
+                entities = self._get_entities(rule)
+                file_entities_map[path] = entities
+        return file_entities_map
+
+    @staticmethod
+    def _get_pattern(rule):
+        """ Convert a rule represented as a dictionary with the keys "domains", "intents",
+        "entities" into a regex pattern.
+
+        Args:
+            rule (dict): Annotation/Unannotation rule.
+
+        Returns:
+            pattern (str): Regex pattern specifying allowed file paths.
+        """
+        pattern = []
+        for x in ["domains", "intents", "files"]:
+            processed_segment = Annotator._process_segment(rule[x])
+            pattern.append(processed_segment)
+        pattern = "/".join(pattern)
+        return ".*/" + pattern
+
+    @staticmethod
+    def _process_segment(segment):
+        """ Process an individual segment from a rule dictionary.
+
+        Args:
+            segment (str): Section of a rule dictionary ("domains", "intents", "entities").
+
+        Returns:
+            segment (str): Cleaned section of the rule dictionary.
+        """
+        segment = re.sub("[()]", "", segment)
+        segment = segment.replace(".*", ".+")
+        segment = segment.replace("*", ".+")
+        segment = segment.split("|")
+        segment = "|".join([x.strip() for x in segment])
```

Comment:
Possible to merge the two lines:
```suggestion
        segment = "|".join([x.strip() for x in segment.split("|")])
```

Are you trying to remove just the newline characters here?
You can simply do the following without the whole (split, strip, join) process:
```suggestion
        segment.replace('\n', '')
```

Reply:
Merged the two lines. This function aims to remove spacing within a regex rule for example, "(  note | chair  )" should be "(note|chair)".

May be removing this function entirely since the current preprocessing of regex could lead to unexpected results when used